### ISSUE TITLE
Heap objects leak and memory usage

### ISSUE THREAD

**Issue by tristanmorgan** [perses/perses#1923]
### What did you do?

Hi,

I am running a single instance of perses with the default file database and a single dashboard and it is experiencing periodic termination due to OOM issues.

### What did you expect to see?

I expect to see a general flattening of the number heap objects and memory usage after a while with a subtle saw tooth pattern due to GC activity.

### What did you see instead? Under which circumstances?

I have noticed the number of objects on the heap correlated with the excess memory used.

![obligatory Perses screenshot](https://github.com/perses/perses/assets/6336384/d325cb48-142b-4705-96d3-e48be70f18c7)

[head of heap dump](https://github.com/perses/perses/files/15031380/perses-steaming-heap.txt)

### System information

Linux 6.1.21-v8+ aarch64

### Perses version

0.44.0

### Perses configuration file

```yaml
database:
  file:
    folder: "/local/perses"
    extension: "json"
        
schemas:
  panels_path: "/etc/perses/cue/schemas/panels"
  queries_path: "/etc/perses/cue/schemas/queries"
  datasources_path: "/etc/perses/cue/schemas/datasources"
  variables_path: "/etc/perses/cue/schemas/variables"
  interval: "5m"
```

### Logs

```text
2024-04-14T07:12:17.008+1000 [ERROR] client.driver_mgr.docker: OOM Killed: container_id=9149b04e72cc99c2943dc8ccd62d18b94e9a1890648bbb553c902fe2d519f830 driver=docker container_id=9149b04e72cc99c2943dc8ccd62d18b94e9a1890648bbb553c902fe2d519f830 container_image=sha256:90665c15d83725d62a8692933d2768d73a0d80d9ee7c70bbd866811491191745 nomad_job_name=perses nomad_task_name=perses nomad_alloc_id=22bcd979-81af-a709-a260-d7f5c5fb82ae
 client.driver_mgr.docker: OOM Killed: container_id=9149b04e72cc99c2943dc8ccd62d18b94e9a1890648bbb553c902fe2d519f830 driver=docker container_id=9149b04e72cc99c2943dc8ccd62d18b94e9a1890648bbb553c902fe2d519f830 container_image=sha256:90665c15d83725d62a8692933d2768d73a0d80d9ee7c70bbd866811491191745 nomad_job_name=perses nomad_task_name=perses nomad_alloc_id=22bcd979-81af-a709-a260-d7f5c5fb82ae
     2024-04-14T07:12:17.011+1000 [INFO]  client.alloc_runner.task_runner: Task event: alloc_id=22bcd979-81af-a709-a260-d7f5c5fb82ae task=perses type=Terminated msg="Exit Code: 137, Exit Message: \"OOM Killed\"" failed=false
  client.alloc_runner.task_runner: Task event: alloc_id=22bcd979-81af-a709-a260-d7f5c5fb82ae task=perses type=Terminated msg="Exit Code: 137, Exit Message: \"OOM Killed\"" failed=false
```

**Comment by celian-garcia** [comment]
Thanks for that report! I will check on our instance to confirm.

**Comment by AntoineThebaud** [comment]
Similar to https://github.com/perses/perses/issues/1440.

**Comment by Nexucis** [comment]
Hello @tristanmorgan thank you for reporting this issue !

After further investigation, I have isolated where the memory leak come from and I have completed an already opened issue on Cuelang on this topic: https://github.com/cue-lang/cue/issues/2121#issuecomment-2066823699

I hope they will look at it, let see.

**Comment by tristanmorgan** [comment]
Thanks everyone for your reply. Let’s hope upstream finds a fix so we all can benefit.

**Comment by Nexucis** [comment]
a workaround would be to increase the frequency of refresh of the cue schemas. Like instead of 5m, setting 1h or 1d would delay by a lot this issue

**Comment by Nexucis** [comment]
At the end, we decided to deactivate the cron. That will mitigate this memory leak until there is a fix and / or until we have finish the plugin flow (#1543)

**Comment by Nexucis** [comment]
This issue should be solved by upgrading the cue dependency to v0.9.0 when available.
